client/signup/steps/site-picker/index.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import SiteSelector from 'components/site-selector';
+import StepWrapper from 'signup/step-wrapper';
+import { getSites } from 'state/selectors';
+import sitesFactory from 'lib/sites-list';
+import { setSelectedSiteId } from 'state/ui/actions';
+import SignupActions from 'lib/signup/actions';
+
+const sites = sitesFactory();
+
+class SitePicker extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.handleSiteSelect = this.handleSiteSelect.bind( this );
+	}
+
+	handleSiteSelect( siteSlug ) {
+		const {
+				stepSectionName,
+				stepName,
+				goToStep,
+				goToNextStep,
+			} = this.props,
+			site = sites.getSite( siteSlug ),
+			hasPlan = site && site.plan && site.plan.product_slug !== 'free_plan';
+
+		this.props.setSelectedSite( site.ID );
+
+		SignupActions.submitSignupStep(
+			{
+				stepName,
+				stepSectionName,
+				siteId: site.ID,
+				siteSlug: site.slug
+			},
+			[],
+			{}
+		);
+
+		if ( hasPlan ) {
+			SignupActions.submitSignupStep( { stepName: 'themes', wasSkipped: true }, [], {
+				themeSlugWithRepo: 'pub/twentysixteen'
+			} );
+			SignupActions.submitSignupStep( { stepName: 'plans', wasSkipped: true }, [], { cartItem: null, privacyItem: null } );
+
+			goToStep( 'user' );
+		} else {
+			goToNextStep();
+		}
+	}
+
+	renderScreen() {
+		return (
+			<Card className="site-picker__wrapper">
+				<SiteSelector
+					filter={
+						function( site ) {
+							return ! site.jetpack;
```

Comment:
A possible additional criteria - being an admin on the site :) Otherwise, they user won't be able to manage the domain, right?